clippy_lints/src/lib.rs

```diff
@@ -20,6 +20,7 @@
 
 // FIXME: switch to something more ergonomic here, once available.
 // (Currently there is no way to opt into sysroot crates without `extern crate`.)
+extern crate indexmap;
```

Comment:
Adding `indexmap` as a dep should be ok, since rustc also uses this dep, so we shouldn't see conflicts with rustc.

But what is the reason to use indexmap from the sysroot rather than a dep? (Or formulated a different way: Why does rustc-dev contain indexmap?)

Reply:
`rustc_data_structures` exports `FxIndexMap`, but not it's matching entry type.

Reply:
I was using `Entry` with `FxIndexMap` which requires using the same `indexmap` dependency as rustc does. Not a problem anymore as the order isn't needed anymore.